cli/admin/src/main/java/io/pravega/cli/admin/dataRecovery/StorageListSegmentsCommand.java

```diff
@@ -0,0 +1,136 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.cli.admin.dataRecovery;
+
+import io.pravega.cli.admin.CommandArgs;
+import io.pravega.common.concurrent.ExecutorServiceHelpers;
+import io.pravega.segmentstore.contracts.SegmentProperties;
+import io.pravega.segmentstore.storage.Storage;
+import io.pravega.segmentstore.storage.StorageFactory;
+import io.pravega.shared.NameUtils;
+import io.pravega.shared.segment.SegmentToContainerMapper;
+import lombok.Cleanup;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.List;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.logging.Level;
+
+/**
+ * Lists all non-shadow segments from there from the storage. The storage is loaded using the config properties.
+ */
+public class StorageListSegmentsCommand extends DataRecoveryCommand {
+    /**
+     * Header line for writing segments' details to csv files.
+     */
+    private static final List<String> HEADER = Arrays.asList("Sealed Status", "Length", "Segment Name");
+    private static final int CONTAINER_EPOCH = 1;
+    private final int containerCount;
+    private final ScheduledExecutorService scheduledExecutorService = ExecutorServiceHelpers.newScheduledThreadPool(10,
+            "listSegmentsProcessor");
+    private final SegmentToContainerMapper segToConMapper;
+    private final StorageFactory storageFactory;
+    private final FileWriter[] csvWriters;
+    private String filePath;
+
+    /**
+     * Creates an instance of StorageListSegmentsCommand class.
+     *
+     * @param args The arguments for the command.
+     */
+    public StorageListSegmentsCommand(CommandArgs args) {
+        super(args);
+        this.containerCount = getServiceConfig().getContainerCount();
+        this.segToConMapper = new SegmentToContainerMapper(this.containerCount);
+        this.storageFactory = createStorageFactory(scheduledExecutorService);
+        this.csvWriters = new FileWriter[this.containerCount];
+    }
+
+    /**
+     * Creates a csv file for each container. All segments belonging to a containerId have their details written to the
+     * csv file for that container.
+     *
+     * @throws Exception   When failed to create/delete file(s).
+     */
+    private void createCSVFiles() throws Exception {
```

Comment:
Why do we need to write all this information in CSV files? Are they going to be used later on for data migration purposes, right?